Clear "redefinition of class UC1701" error

crasbe opened the pull request:
Depending on the program structure, header files can be included more than once in the program code. This means that the class would be redefined, which results in a compiler error. To avoid this, I added this standard macro to make the preprocessor add the code only once and therefore avoid the said compiler error.

Compiler error:
"error: redefinition of 'class UC1701'
 class UC1701: public Print {"

Industruino commented:
Thank you Nick, it is much appreciated! Great to have active customers like you contributing to the Industruino community.
I will merge the patches into the main branch. 

Thanks again. 
Cheers,
Loic

On 5 Jul 2015, at 21:34, Nicholas Humfrey <notifications@github.com> wrote:

> This Pull Request add support for CR and LF characters being written to the LCD.
> 
> This means that lcd.println() works correctly, making it easier to write multiple lines of text with using lcd.setCursor() between each lcd.print().
> 
> The disadvantage is that you can no-longer use characters 10 and 13 for user-defined glyphs.
> 
> So far I am only using 2 custom glyphs in my project anyway.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/Industruino/UC1701/pull/3
> 
> Commit Summary
> 
> Added support writing CR and LF to LCD
> File Changes
> 
> M UC1701.cpp (18)
> Patch Links:
> 
> https://github.com/Industruino/UC1701/pull/3.patch
> https://github.com/Industruino/UC1701/pull/3.diff
> —
> Reply to this email directly or view it on GitHub.
>

njh commented:
Any news on merging this?

Industruino commented:
Thanks Nick, somehow I overlooked this pull request. Thank you for contributing!

crasbe committed:
Clear "redefinition of class UC1701" error

Depending on the program structure, header files can be included more than once in the program code. This means that the class would be redefined, which results in a compiler error. To avoid this, I added this standard macro to make the preprocessor add the code only once and therefore avoid the said compiler error.

Compiler error:
"error: redefinition of 'class UC1701'
 class UC1701: public Print {"

Industruino commented:
@crasbe thank you!
